In [1]:
import numpy as np
import pandas as pd

In [2]:
# The statistical module used to run chi square test
import scipy.stats as stats

In [3]:
# Observed data in a (hypothetical) survey of 300 people 
observed = pd.Series([9,8,16,10,8,47], index=["al","ar","la","ms","ok","tx"])

In [4]:
# Create a data frame
df = pd.DataFrame([observed]).T

In [5]:
# Add a column whose default values are the expected values
df[1] = 29.8

In [6]:
# Rename columns
df.columns = ["observed", "expected"]

In [8]:
# View the data frame
df

observed  expected
al         9      29.8
ar         8      29.8
la        16      29.8
ms        10      29.8
ok         8      29.8
tx        47      29.8

In [9]:
# The degree of freedom is 3-1 = 2
# With a p-value of 0.05, the confidence level is 1.00-0.05 = 0.95.
critical_value = stats.chi2.ppf(q = 0.95, df = 5)

In [10]:
# The critical value
critical_value

11.070497693516351

In [11]:
# Run the chi square test with stats.chisquare()
stats.chisquare(df['observed'], df['expected'])

Power_divergenceResult(statistic=75.88724832214764, pvalue=6.073209206643199e-15)

### Conclusion

* Since the chi square value of 75.89 exceeds the critical value of 11.07, we conclude that the results are statistically significant.